packages/greencheck/src/Sitecheck.php

```diff
@@ -83,42 +84,66 @@ class Sitecheck
     protected $_countrytlds = null;
 
     /**
-     * @var Table\GreencheckUrl
+     * @var GreencheckUrlRepository
      */
     protected $greencheckUrl = null;
 
     /**
-     * @var Table\GreencheckIp
+     * @var GreencheckIpRepository
      */
     protected $greencheckIp = null;
 
     /**
-     * @var Table\GreencheckAs
+     * @var GreencheckAsRepository
      */
     protected $greencheckAs = null;
 
     protected $aschecker = null;
 
+    /**
+     * @var GreencheckTldRepository
+     */
+    private $greencheckTld;
+    /**
+     * @var Logger
+     */
+    private $logger;
+
     /**
      * Construct the sitecheck.
      *
-     * @param EntityManager   $em         Doctrine Entity Manager
-     * @param Sitecheck/Cache $cache      Cache object
-     * @param string          $calledfrom [description]
+     * @param GreencheckUrlRepository $greencheckUrlRepository
+     * @param GreencheckIpRepository  $greencheckIpRepository
+     * @param GreencheckAsRepository  $greencheckAsRepository
+     * @param GreencheckTldRepository $greencheckTldRepository
+     * @param Cache $cache
+     * @param string $calledfrom [description]
      */
-    public function __construct($em, $cache, $calledfrom = 'website')
-    {
+    public function __construct(
+        GreencheckUrlRepository $greencheckUrlRepository,
+        GreencheckIpRepository $greencheckIpRepository,
+        GreencheckAsRepository $greencheckAsRepository,
+        GreencheckTldRepository $greencheckTldRepository,
+        Cache $cache,
+        Logger $logger,
+        $calledfrom = 'website',
+        DnsFetcher $dnsFetcher = null
+    ) {
         $this->_calledfrom = $calledfrom;
-        $this->em = $em;
 
         $this->validator = new Validator();
+        if (null == $dnsFetcher) {
+            $dnsFetcher = new DnsFetcher();
+        }
+        $this->dnsFetcher = $dnsFetcher;
 
-        $this->greencheckUrl = $this->em->getRepository("TGWF\Greencheck\Entity\GreencheckUrl");
-        $this->greencheckIp = $this->em->getRepository("TGWF\Greencheck\Entity\GreencheckIp");
-        $this->greencheckAs = $this->em->getRepository("TGWF\Greencheck\Entity\GreencheckAs");
-        $this->greencheckTld = $this->em->getRepository("TGWF\Greencheck\Entity\GreencheckTld");
+        $this->greencheckUrl = $greencheckUrlRepository;
```

Comment:
rename to $this->greencheckUrlRepository (same for 3 below and in the rest of the class)